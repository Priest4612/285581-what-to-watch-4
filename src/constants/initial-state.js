export default {
  error: null,
  loading: true,
  movies: [],
  movieIdActive: null,
  genres: [],
  genreIdAcrive: null,
};
