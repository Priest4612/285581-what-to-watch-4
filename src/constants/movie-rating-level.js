const MovieRatingLevel = {
  BAD: {
    NAME: `Bad`,
    RATING: {
      MIN: 0,
      MAX: 3,
    }
  },
  NORMAL: {
    NAME: `Normal`,
    RATING: {
      MIN: 3,
      MAX: 5,
    }
  },
  GOOD: {
    NAME: `Good`,
    RATING: {
      MIN: 5,
      MAX: 8,
    }
  },
  VERY_GOOD: {
    NAME: `Very good`,
    RATING: {
      MIN: 8,
      MAX: 10,
    }
  },
  AWESOME: {
    NAME: `Awesome`,
    RATING: {
      MIN: 10,
      MAX: 10,
    }
  },
};

export {MovieRatingLevel};
