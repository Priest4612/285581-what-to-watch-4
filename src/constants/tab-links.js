const tabList = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export {tabList};
