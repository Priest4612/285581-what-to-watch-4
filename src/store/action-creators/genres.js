import ActionType from '../action-type';

const genresLoaded = (genres) => ({
  type: ActionType.Genres.FETCH_GENRES_SUCCESS,
  payload: genres,
});

const setActiveGenre = (name) => ({
  type: ActionType.Genres.CHANGE_ACTIVE_GENRE,
  payload: name,
});

export default {
  genresLoaded,
  setActiveGenre,
};
