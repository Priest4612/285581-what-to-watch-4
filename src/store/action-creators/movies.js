import ActionType from '../action-type';

const moviesLoaded = (movies) => ({
  type: ActionType.Movies.FETCH_MOVIES_SUCCESS,
  payload: movies,
});

const setActiveMovieId = (id) => ({
  type: ActionType.Movies.CHANGE_ACTIVE_MOVIE,
  payload: id,
});

export default {
  moviesLoaded,
  setActiveMovieId,
};
