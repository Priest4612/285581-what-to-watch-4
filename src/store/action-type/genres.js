export default {
  FETCH_GENRES_FAILURE: `FETCH_GENRES_FAILURE`,
  FETCH_GENRES_SUCCESS: `FETCH_GENRES_SUCCESS`,
  FETCH_GENRES_REQUEST: `FETCH_MOVIES_REQUEST`,
  CHANGE_ACTIVE_GENRE: `CHANGE_ACTIVE_GENRE`,
};
