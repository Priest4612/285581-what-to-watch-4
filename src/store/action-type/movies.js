export default {
  FETCH_MOVIES_FAILURE: `FETCH_MOVIES_FAILURE`,
  FETCH_MOVIES_SUCCESS: `FETCH_MOVIES_SUCCESS`,
  FETCH_MOVIES_REQUEST: `FETCH_MOVIES_REQUEST`,
  CHANGE_ACTIVE_MOVIE: `CHANGE_ACTIVE_MOVIE`,
};
