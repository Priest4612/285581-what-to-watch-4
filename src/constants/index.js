import {ViewMode} from './view-mode.js';
import {MovieRatingLevel} from './movie-rating-level.js';
import {tabList} from './tab-links.js';

export {
  ViewMode,
  MovieRatingLevel,
  tabList,
};
