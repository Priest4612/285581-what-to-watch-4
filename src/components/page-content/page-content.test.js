import React from 'react';
import renderer from 'react-test-renderer';

import movies from '../../mocks/test-films-mock.json';
import genres from '../../mocks/test-genres-mock.json';
import {ViewMode} from '../../constants';

import PageContent from './page-content.jsx';


it(`Render PageContent -- ViewMode.MAIN`, () => {
  const tree = renderer
    .create(<PageContent
      onClickCard={() => {}}
      onMouseEnterCard={() => {}}
      onMouseLeaveCard={() => {}}
      onClickGenreItem={() => {}}
      movieDetails={movies[0]}
      viewMode={ViewMode.MAIN}
      genres={genres}
      movies={movies}
      isMuted={true}
      isPlaying={false}
      activeCardId={2}
      activeGenre={genres[3].name}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render PageContent  -- ViewMode.FULL`, () => {
  const tree = renderer
    .create(<PageContent
      onClickCard={() => {}}
      onMouseEnterCard={() => {}}
      onMouseLeaveCard={() => {}}
      onClickGenreItem={() => {}}
      movieDetails={movies[0]}
      viewMode={ViewMode.FULL}
      genres={genres}
      movies={movies}
      isMuted={true}
      isPlaying={false}
      activeCardId={2}
      activeGenre={genres[3].name}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
