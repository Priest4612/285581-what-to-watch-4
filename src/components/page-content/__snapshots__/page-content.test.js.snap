// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PageContent  -- ViewMode.FULL 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog catalog--like-this"
  >
    <h2
      className="catalog__title"
    >
      More like this
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="We Need To Talk About Kevin"
            height="175"
            src="img/we-need-to-talk-about-kevin.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
          >
            We Need To Talk About Kevin
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Render PageContent -- ViewMode.MAIN 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <ul
      className="catalog__genres-list"
    >
      <li
        className="catalog__genres-item
            false"
        onClick={[Function]}
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Все жанры
        </a>
      </li>
      <li
        className="catalog__genres-item
            false"
        onClick={[Function]}
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Криминал
        </a>
      </li>
      <li
        className="catalog__genres-item
            false"
        onClick={[Function]}
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Исторический
        </a>
      </li>
      <li
        className="catalog__genres-item
            catalog__genres-item--active"
        onClick={[Function]}
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Триллер
        </a>
      </li>
      <li
        className="catalog__genres-item
            false"
        onClick={[Function]}
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Короткометражный
        </a>
      </li>
      <li
        className="catalog__genres-item
            false"
        onClick={[Function]}
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Мистика
        </a>
      </li>
      <li
        className="catalog__genres-item
            false"
        onClick={[Function]}
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Вестерн
        </a>
      </li>
      <li
        className="catalog__genres-item
            false"
        onClick={[Function]}
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Ужасы
        </a>
      </li>
      <li
        className="catalog__genres-item
            false"
        onClick={[Function]}
      >
        <a
          className="catalog__genres-link"
          href="#"
        >
          Спорт
        </a>
      </li>
    </ul>
    <div
      className="catalog__movies-list"
    />
    <div
      className="catalog__more"
    >
      <button
        className="catalog__button"
        type="button"
      >
        Show more
      </button>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
